### PR TITLE
add namespace module

### PR DESCRIPTION
**Pull request by AlexanderBrevig**
Add a namespace module, this will be referred to in upcoming blog post about branch environments.

### CODE REVIEW

File: modules/namespace/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+If you specify a reserved namespace such as `default`, `dev`, `staging` or `production` no namespace will be created.
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| labels | The labels you wish to decorate with | string | n/a | yes |
+| labels.team | The name of your team or department | string | n/a | yes |
+| labels.app | The name of this application / workload | string | n/a | yes |
```
Comment (PavloRPsiuk) [inline modules/namespace/README.md:19]: As I understand it can be more than one application in a namespace.
I don't know if you should add slack and environment labels :thinking:

File: examples/namespace-defaults/main.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# This will end up not creating a namespace
+# because we target master branch
+module "namespace" {
+  #source = "github.com/entur/terraform//modules/namespace"
```
Comment (sindrel) [inline examples/namespace-defaults/main.tf:9]: Examples should probably reference the public module path, instead of the local/dev path

File: modules/namespace/README.md (hunk 1 above)
Comment (AlexanderBrevig) [reply modules/namespace/README.md:19]: The label spec is not really documented in full for all modules. The other modules only document the ones needed, and we should probably start to encourage people to use the Entur module, which in fact checks for all required labels.

TL;DR this module does not require any labels. Should I remove l18 and l19 and leave l17?

**Comment by sindrel** [comment]
I'm a bit uncertain about the terms used. It might be the examples throwing me off where there seems to be an unclear mix of namespaces and git branches, i.e.:

```
variable "kubernetes_namespace" {
  description = "Your kubernetes namespace"
  default     = "master"
}
```

* Could there be clearer differentiation between when or where to provide a Git branch name vs. a Kubernetes namespace?
* The assumption that the main Git branch is named master could be a bad idea as it could be anything set by the team, and also not unlikely to change (https://github.com/github/renaming)
* Should we add a label marking that it's a feature based namespace in addition to the timestamp?

Also I'm thinking we should probably use the module to set a convention:

All feature based namespaces could be prefixed with something recognizable so that we know it's ephemeral, and to make them easily sortable/identifiable, such as "feature-" or "dev-". 

Perhaps only allow feature namespaces based on the origin git branch name matching a certain rule, such as "feature-*", "dev-*"?

**Comment by AlexanderBrevig** [comment]
Maybe the API here should be simpler and not imply any conventions, and the resolution of branch vs namespace should be handled elsewhere?

That is, I do not rewrite the input `kubernetes_namespace` so the callsite needs to handle it however they see fit?
